src/couch_replicator_scheduler.erl

```diff
@@ -205,52 +213,62 @@ handle_config_terminate(Self, _, _) ->
 
 %% private functions
 
-start_jobs(Count) ->
+start_jobs(Count, MaxHistory) ->
     Runnable0 = pending_jobs(),
     Runnable1 = lists:sort(fun oldest_job_first/2, Runnable0),
     Runnable2 = lists:filter(fun not_recently_crashed/1, Runnable1),
     Runnable3 = lists:sublist(Runnable2, Count),
-    lists:foreach(fun start_job_int/1, Runnable3).
+    lists:foreach(fun(J) -> start_job_int(J, MaxHistory) end, Runnable3).
 
 
-stop_jobs(Count) ->
+stop_jobs(Count, MaxHistory) ->
     Running0 = running_jobs(),
     Running1 = lists:sort(fun oldest_job_first/2, Running0),
     Running2 = lists:sublist(Running1, Count),
-    lists:foreach(fun stop_job_int/1, Running2).
+    lists:foreach(fun(J) -> stop_job_int(J, MaxHistory) end, Running2).
 
 
 oldest_job_first(#job{} = A, #job{} = B) ->
     last_started(A) =< last_started(B).
 
 
 not_recently_crashed(#job{} = Job) ->
-    case crash_history(Job) of
+    case Job#job.history of
         [] ->
             true;
-        [{crashed, When} | _] ->
-            timer:now_diff(os:timestamp(), When)
-                >= ?MINIMUM_CRASH_INTERVAL
+        [stopped | _] ->
+            true;
+        _ ->
+            EventsBeforeStop = lists:takewhile(
+                fun({Event, _}) -> Event =/= stopped end, Job#job.history),
+            Crashes = lists:filter(
+                fun({{crashed, _Reason}, _When}) -> true; (_) -> false end,
+                EventsBeforeStop
+            ),
```

Comment:
The comprehension was getting a bit long, and I prefer multiline funs rather than multiline list comprehensions, but I could change it back to a long list comprehension.